code-challenges.test.js

```diff
@@ -14,38 +14,116 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
-
+describe ("namesCap", () => {
+  it("takes an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const desc = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }]
+    expect(namesCap(desc)).toEqual("Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee.")
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+// namesCap
+
+//   ✕ takes an array of objects and returns an array with a sentence about each person with their name capitalized (1 ms)
+// ● namesCap › takes an array of objects and returns an array with a sentence about each person with their name capitalized
+//   ReferenceError: namesCap is not defined
 
 // b) Create the function that makes the test pass.
 
-
+//PseudoCode:
+
+// Return an array with a sentence about each person with their name capitalized.
+// Use .map to create a new array calling on the provided function.
+// Use .name to indicate the function name specified.
+// In order to split the string into an array of substrings use .split.
+// To replace the regular expression use .replace.
+// And finally use .toUpperCase to convert the the names to uppercase letters.
+
+  const namesCap = (array) => {
+    return array.map(newSentence => {
+      let name = newSentence.name.split(" ").map(value => value.replace(value[0], value[0].toUpperCase()))
+      let fullname = name.join(" ")
+      let desc = desc.occupation
+      let newSentence = `${fullname} is ${desc}.`
+      return newSentence
+    })
+  }
+
```

Comment:
You are so close. If you do corrections according to the error code, you will be able to pass.

Error messages:
`SyntaxError: /Users/charleanbaxter/Desktop/learn-2022-instruction/lecture-2022/react-state-cb/assessment.test.js: Identifier 'newSentence' has already been declared. (272:19)`
Correction ----> Changed the variable to a different name

 let sentence = `${fullname} is ${desc}.`
 return sentence

`ReferenceError: Cannot access 'desc' before initialization`
Correction ----> change the reference to the parameter designated on your function

let desc = newSentence.occupation

Further inspection shows that the expected outcome on the test was inputted without the square brackets to show the array.

    expect(namesCap(desc)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])

After the update, you passed. Follow the error messages, they are your friends.

`namesCap
    ✓ takes an array of objects and returns an array with a sentence about each person with their name capitalized (1 ms)`